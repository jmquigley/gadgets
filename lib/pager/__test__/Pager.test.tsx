'use strict';

import * as assert from 'assert';
import {mount, shallow} from 'enzyme';
import * as React from 'react';
// import {waitPromise} from 'util.wait';
import {getDefaultPagerProps, Pager} from '../index';

function validate(ctl: any) {
	assert(ctl);
	assert(!ctl.prop('disabled'));
	assert(ctl.prop('visible'));
	assert.equal(ctl.find('.ui-pager').length, 1);
}

test('Test retrieval of Pager props object', () => {
	const props = getDefaultPagerProps();

	assert(props);
	expect(props).toMatchSnapshot();
});

test('Test creation of a Pager control', () => {
	const ctl = shallow(<Pager className="test-class" testing/>);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test pager getPages method with front of list', () => {
	const ctl = mount(
		<Pager
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.deepEqual(pager.pages, [1, 2, 3]);
	assert.equal(pager.currentPage, 1);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);
	assert.equal(pager.state.pageSize, pager.pageSize);
});

test('Test pager getPages method with 2nd to last in list', () => {
	const ctl = mount(
		<Pager
			initialPage={11}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.deepEqual(pager.pages, [10, 11, 12]);
	assert.equal(pager.currentPage, 11);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);
	assert.equal(pager.state.pageSize, pager.pageSize);
});

test('Test Pager getPages method with last in the List', () => {
	const ctl = mount(
		<Pager
			initialPage={12}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.deepEqual(pager.pages, [10, 11, 12]);
	assert.equal(pager.currentPage, 12);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);
	assert.equal(pager.state.pageSize, pager.pageSize);
});

test('Test Pager getPages method with invalid initial page (negative test)', () => {
	const ctl = mount(
		<Pager
			initialPage={-1}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.deepEqual(pager.pages, [1, 2, 3]);
	assert.equal(pager.currentPage, 1);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);
	assert.equal(pager.state.pageSize, pager.pageSize);
});

test('Test Pager getPages method with invalid pageSizes (negative test)', () => {
	const ctl = mount(
		<Pager
			pageSizes={null}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.deepEqual(pager.pages, [1, 2, 3]);
	assert.equal(pager.currentPage, 1);
	assert.equal(pager.lastPage, 12);
	assert.equal(ctl.state('pageSize'), 25);
	assert.equal(pager.state.pageSize, pager.pageSize);
});

test('Test Pager getPages method with empty pageSizes (negative test)', () => {
	const ctl = mount(
		<Pager
			pageSizes={[]}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.deepEqual(pager.pages, [1, 2, 3]);
	assert.equal(pager.currentPage, 1);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);
	assert.equal(pager.state.pageSize, pager.pageSize);
});

test('Test Pager getPages method with invalid totalItems (negative test)', () => {
	const ctl = mount(
		<Pager
			pageSizes={[]}
			testing
			totalItems={0}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.deepEqual(pager.pages, [1, 0, 0]);
	assert.equal(pager.currentPage, 1);
	assert.equal(pager.lastPage, 1);
	assert.equal(pager.state.pageSize, 25);
	assert.equal(pager.state.pageSize, pager.pageSize);
});

test('Test selection of the second page in the control', () => {
	const select = jest.fn();
	const ctl = mount(
		<Pager
			onSelect={select}
			pageSizes={[10, 25, 100]}
			testing
			totalItems={3000}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	// Number of button controls in this instance
	assert.equal(ctl.find('.ui-button').length, 5);
	assert.equal(ctl.find('.ui-button-text').length, 3);
	assert.equal(ctl.find('.ui-button-dialog').length, 1);

	// Click the second page number.  Can't click the first one because
	// it is already selected by default when the control is created
	ctl.find('.ui-button-text').at(1).simulate('click');
	expect(select).toHaveBeenCalled();
	assert.equal(select.mock.calls[0][0], 2);
});

test('Test pressing the "<<" (first) button', () => {
	const select = jest.fn();
	const ctl = mount(
		<Pager
			initialPage={2}
			onSelect={select}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.equal(pager.currentPage, 2);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);

	// Number of button controls in this instance
	assert.equal(ctl.find('.ui-button').length, 5);
	assert.equal(ctl.find('.ui-button-text').length, 3);
	assert.equal(ctl.find('.ui-button-dialog').length, 1);

	// Select the first button "<<" move from 2 -> 1
	ctl.find('.ui-button').at(0).simulate('click');
	expect(select).toHaveBeenCalled();
	assert.equal(select.mock.calls[0][0], 1);
});

test('Test pressing the "<" (previous) button', () => {
	const select = jest.fn();
	const ctl = mount(
		<Pager
			initialPage={3}
			onSelect={select}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.equal(pager.currentPage, 3);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);

	// Number of button controls in this instance
	assert.equal(ctl.find('.ui-button').length, 5);
	assert.equal(ctl.find('.ui-button-text').length, 3);
	assert.equal(ctl.find('.ui-button-dialog').length, 1);

	// Select the first button "<" move from 3 -> 2
	ctl.find('.ui-button').at(1).simulate('click');
	expect(select).toHaveBeenCalled();
	assert.equal(select.mock.calls[0][0], 2);
});

test('Test pressing the ">" (next) button', () => {
	const select = jest.fn();
	const ctl = mount(
		<Pager
			onSelect={select}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.equal(pager.currentPage, 1);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);

	// Number of button controls in this instance
	assert.equal(ctl.find('.ui-button').length, 5);
	assert.equal(ctl.find('.ui-button-text').length, 3);
	assert.equal(ctl.find('.ui-button-dialog').length, 1);

	// Select the third button ">", move from 1 -> 2
	ctl.find('.ui-button').at(2).simulate('click');
	expect(select).toHaveBeenCalled();
	assert.equal(select.mock.calls[0][0], 2);
});

test('Test pressing the ">>" (last) button', () => {
	const select = jest.fn();
	const ctl = mount(
		<Pager
			onSelect={select}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);

	assert.equal(pager.currentPage, 1);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);

	// Number of button controls in this instance
	assert.equal(ctl.find('.ui-button').length, 5);
	assert.equal(ctl.find('.ui-button-text').length, 3);
	assert.equal(ctl.find('.ui-button-dialog').length, 1);

	// Select the fourth button ">>", move from 1 -> 12
	ctl.find('.ui-button').at(3).simulate('click');
	expect(select).toHaveBeenCalled();
	assert.equal(select.mock.calls[0][0], 12);
});

// FIXME: this test no longer works as expected after enzyme breaking changes
/*
test('Test selecting dialog "50" to change the page size', async () => {
	const select = jest.fn();
	const ctl = mount(
		<Pager
			onSelect={select}
			testing
			totalItems={299}
		/>
	);

	validate(ctl);
	const pager = ctl.instance() as Pager;
	assert(pager);
	assert(pager.dialog);

	assert.equal(pager.currentPage, 1);
	assert.equal(pager.lastPage, 12);
	assert.equal(pager.state.pageSize, 25);

	// Number of button controls in this instance
	assert.equal(ctl.find('.ui-button').length, 5);
	assert.equal(ctl.find('.ui-button-text').length, 3);
	assert.equal(ctl.find('.ui-button-dialog').length, 1);
	assert.equal(ctl.find('.ui-title').length, 16);

	// Select the "50" from the dialog list, click and check page size
	// Goes from 25 -> 50
	ctl.find('.ui-title').at(5).simulate('click');

	// This wait must occur during the test because there is a built in click
	// delay where the component checks if a double click is occurring.
	await waitPromise(1)
		.then(() => {
			expect(select).toHaveBeenCalled();
			assert.equal(select.mock.calls[0][0], 2);

			assert.equal(pager.currentPage, 1);
			assert.equal(pager.lastPage, 6);
			assert.equal(pager.state.pageSize, 50);
		})
		.catch((err: string) => {
			assert.fail(err);
		});
});
*/

test('Repeatedly create instance with different initial start for props test', () => {
	for (let i = 1; i < 5; i++) {
		const ctl = mount(
			<Pager
				initialPage={i}
				testing
				totalItems={299}
			/>
		);

		validate(ctl);
		const pager = ctl.instance() as Pager;
		assert(pager);

		assert.equal(pager.currentPage, i);
	}
});

// TODO: add a test to show changing page size but resetting current to 1
// TODO: add a test to show changing page size keeps current page
// TODO: add test case for TextInput in Pager
// TODO: add test for Pager blur, change, keyPress
// TODO: add test to show pressing the sort dialog buttons
