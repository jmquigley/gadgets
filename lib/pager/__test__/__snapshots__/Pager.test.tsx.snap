// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a Pager control 1`] = `
<div
  className="ui-pager pager normal test-class"
>
  <Button
    backgroundColor="inherit"
    borderColor="inherit"
    borderWidth="none"
    className="pagerButton normalBox"
    color="inherit"
    contentEditable={false}
    direction="right"
    disabled={false}
    focus={false}
    height=""
    iconName="angle-double-left"
    iconStyle=""
    id=""
    initialPage={1}
    initialPageSize={25}
    location="none"
    noedit={false}
    nohover={false}
    noripple={false}
    onChangePageSize={[Function]}
    onClick={[Function]}
    onSelect={[Function]}
    onSort={null}
    pageSizes={
      Array [
        25,
        50,
        100,
      ]
    }
    pagesToDisplay={3}
    selected={false}
    sizing="normal"
    style={Object {}}
    totalItems={0}
    useinput={false}
    visible={true}
    width=""
  />
  <Button
    backgroundColor="inherit"
    borderColor="inherit"
    borderWidth="none"
    className="pagerButton normalBox"
    color="inherit"
    contentEditable={false}
    direction="right"
    disabled={false}
    focus={false}
    height=""
    iconName="angle-left"
    iconStyle=""
    id=""
    initialPage={1}
    initialPageSize={25}
    location="none"
    noedit={false}
    nohover={false}
    noripple={false}
    onChangePageSize={[Function]}
    onClick={[Function]}
    onSelect={[Function]}
    onSort={null}
    pageSizes={
      Array [
        25,
        50,
        100,
      ]
    }
    pagesToDisplay={3}
    selected={false}
    sizing="normal"
    style={Object {}}
    totalItems={0}
    useinput={false}
    visible={true}
    width=""
  />
  <ButtonText
    backgroundColor="inherit"
    borderColor="inherit"
    borderWidth="none"
    className="pagerButton normalBox ui-pager-selected"
    color="inherit"
    contentEditable={false}
    direction="right"
    disabled={false}
    focus={false}
    height=""
    iconName="bomb"
    id=""
    imageFile=""
    initialPage={1}
    initialPageSize={25}
    justify={1}
    location="none"
    noedit={false}
    nohover={false}
    noicon={true}
    noripple={false}
    onChangePageSize={[Function]}
    onClick={[Function]}
    onSelect={[Function]}
    onSort={null}
    pageSizes={
      Array [
        25,
        50,
        100,
      ]
    }
    pagesToDisplay={3}
    selected={false}
    sizing="normal"
    style={Object {}}
    text="1"
    totalItems={0}
    useinput={false}
    visible={true}
    width=""
  />
  <ButtonText
    backgroundColor="inherit"
    borderColor="inherit"
    borderWidth="none"
    className="pagerButton normalBox"
    color="inherit"
    contentEditable={false}
    direction="right"
    disabled={true}
    focus={false}
    height=""
    iconName="bomb"
    id=""
    imageFile=""
    initialPage={1}
    initialPageSize={25}
    justify={1}
    location="none"
    noedit={false}
    nohover={false}
    noicon={true}
    noripple={false}
    onChangePageSize={[Function]}
    onClick={[Function]}
    onSelect={[Function]}
    onSort={null}
    pageSizes={
      Array [
        25,
        50,
        100,
      ]
    }
    pagesToDisplay={3}
    selected={false}
    sizing="normal"
    style={Object {}}
    text=""
    totalItems={0}
    useinput={false}
    visible={true}
    width=""
  />
  <ButtonText
    backgroundColor="inherit"
    borderColor="inherit"
    borderWidth="none"
    className="pagerButton normalBox"
    color="inherit"
    contentEditable={false}
    direction="right"
    disabled={true}
    focus={false}
    height=""
    iconName="bomb"
    id=""
    imageFile=""
    initialPage={1}
    initialPageSize={25}
    justify={1}
    location="none"
    noedit={false}
    nohover={false}
    noicon={true}
    noripple={false}
    onChangePageSize={[Function]}
    onClick={[Function]}
    onSelect={[Function]}
    onSort={null}
    pageSizes={
      Array [
        25,
        50,
        100,
      ]
    }
    pagesToDisplay={3}
    selected={false}
    sizing="normal"
    style={Object {}}
    text=""
    totalItems={0}
    useinput={false}
    visible={true}
    width=""
  />
  <Button
    backgroundColor="inherit"
    borderColor="inherit"
    borderWidth="none"
    className="pagerButton normalBox"
    color="inherit"
    contentEditable={false}
    direction="right"
    disabled={false}
    focus={false}
    height=""
    iconName="angle-right"
    iconStyle=""
    id=""
    initialPage={1}
    initialPageSize={25}
    location="none"
    noedit={false}
    nohover={false}
    noripple={false}
    onChangePageSize={[Function]}
    onClick={[Function]}
    onSelect={[Function]}
    onSort={null}
    pageSizes={
      Array [
        25,
        50,
        100,
      ]
    }
    pagesToDisplay={3}
    selected={false}
    sizing="normal"
    style={Object {}}
    totalItems={0}
    useinput={false}
    visible={true}
    width=""
  />
  <Button
    backgroundColor="inherit"
    borderColor="inherit"
    borderWidth="none"
    className="pagerButton normalBox"
    color="inherit"
    contentEditable={false}
    direction="right"
    disabled={false}
    focus={false}
    height=""
    iconName="angle-double-right"
    iconStyle=""
    id=""
    initialPage={1}
    initialPageSize={25}
    location="none"
    noedit={false}
    nohover={false}
    noripple={false}
    onChangePageSize={[Function]}
    onClick={[Function]}
    onSelect={[Function]}
    onSort={null}
    pageSizes={
      Array [
        25,
        50,
        100,
      ]
    }
    pagesToDisplay={3}
    selected={false}
    sizing="normal"
    style={Object {}}
    totalItems={0}
    useinput={false}
    visible={true}
    width=""
  />
  <div
    className="spacer"
  />
  <div
    className="spacer"
  />
  <ButtonDialog
    backgroundColor="inherit"
    borderColor="inherit"
    borderWidth="none"
    className="pagerDialog"
    color="inherit"
    contentEditable={false}
    dialogClasses={Array []}
    direction="right"
    disabled={false}
    focus={false}
    height=""
    iconName="ellipsis-v"
    iconStyle=""
    id=""
    initialPage={1}
    initialPageSize={25}
    location="top"
    noedit={false}
    nohover={false}
    noripple={false}
    notriangle={true}
    onChangePageSize={[Function]}
    onClick={[Function]}
    onSelect={[Function]}
    onSort={null}
    pageSizes={
      Array [
        25,
        50,
        100,
      ]
    }
    pagesToDisplay={3}
    selected={false}
    sizing="normal"
    style={Object {}}
    totalItems={0}
    triangleClasses={Array []}
    useinput={false}
    visible={true}
    width=""
  >
    <List
      alternating={false}
      backgroundColor="inherit"
      borderColor="inherit"
      borderWidth="none"
      className=""
      color="inherit"
      contentEditable={false}
      direction="right"
      disabled={false}
      focus={false}
      height=""
      id=""
      location="none"
      noedit={false}
      nohover={false}
      noripple={false}
      onAdd={[Function]}
      selected={false}
      sizing="normal"
      style={Object {}}
      unselect={false}
      visible={true}
      width=""
    >
      <ListItem
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className="test-class"
        color="inherit"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        hiddenLeftButton={false}
        hiddenRightButton={false}
        href={
          Object {
            "selectHandler": [Function],
            "sizing": "normal",
          }
        }
        id=""
        initialPage={1}
        initialPageSize={25}
        layout={5}
        leftButton={
          <Icon
            backgroundColor="inherit"
            borderColor="inherit"
            borderWidth="none"
            className=""
            color="inherit"
            contentEditable={false}
            direction="right"
            disabled={false}
            focus={false}
            height=""
            iconName=""
            id=""
            imageFile=""
            location="none"
            noedit={false}
            nohover={false}
            noripple={false}
            selected={false}
            sizing="normal"
            style={Object {}}
            visible={true}
            width=""
          />
        }
        location="none"
        noedit={true}
        nohover={false}
        noripple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onChangePageSize={[Function]}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseOut={[Function]}
        onSelect={[Function]}
        onSort={null}
        onUpdate={[Function]}
        pageSizes={
          Array [
            25,
            50,
            100,
          ]
        }
        pagesToDisplay={3}
        rightButton={null}
        selected={false}
        sizing="normal"
        stacked={false}
        style={Object {}}
        title="First"
        totalItems={0}
        useedit={false}
        useinput={false}
        visible={true}
        widget={null}
        width=""
      />
      <ListItem
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className="test-class"
        color="inherit"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        hiddenLeftButton={false}
        hiddenRightButton={false}
        href={
          Object {
            "selectHandler": [Function],
            "sizing": "normal",
          }
        }
        id=""
        initialPage={1}
        initialPageSize={25}
        layout={5}
        leftButton={
          <Icon
            backgroundColor="inherit"
            borderColor="inherit"
            borderWidth="none"
            className=""
            color="inherit"
            contentEditable={false}
            direction="right"
            disabled={false}
            focus={false}
            height=""
            iconName=""
            id=""
            imageFile=""
            location="none"
            noedit={false}
            nohover={false}
            noripple={false}
            selected={false}
            sizing="normal"
            style={Object {}}
            visible={true}
            width=""
          />
        }
        location="none"
        noedit={true}
        nohover={false}
        noripple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onChangePageSize={[Function]}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseOut={[Function]}
        onSelect={[Function]}
        onSort={null}
        onUpdate={[Function]}
        pageSizes={
          Array [
            25,
            50,
            100,
          ]
        }
        pagesToDisplay={3}
        rightButton={null}
        selected={false}
        sizing="normal"
        stacked={false}
        style={Object {}}
        title="Last"
        totalItems={0}
        useedit={false}
        useinput={false}
        visible={true}
        widget={null}
        width=""
      />
      <ListItem
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className="test-class"
        color="inherit"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        hiddenLeftButton={false}
        hiddenRightButton={false}
        href={
          Object {
            "selectHandler": [Function],
            "sizing": "normal",
          }
        }
        id=""
        initialPage={1}
        initialPageSize={25}
        layout={5}
        leftButton={
          <Icon
            backgroundColor="inherit"
            borderColor="inherit"
            borderWidth="none"
            className=""
            color="inherit"
            contentEditable={false}
            direction="right"
            disabled={false}
            focus={false}
            height=""
            iconName=""
            id=""
            imageFile=""
            location="none"
            noedit={false}
            nohover={false}
            noripple={false}
            selected={false}
            sizing="normal"
            style={Object {}}
            visible={true}
            width=""
          />
        }
        location="none"
        noedit={true}
        nohover={false}
        noripple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onChangePageSize={[Function]}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseOut={[Function]}
        onSelect={[Function]}
        onSort={null}
        onUpdate={[Function]}
        pageSizes={
          Array [
            25,
            50,
            100,
          ]
        }
        pagesToDisplay={3}
        rightButton={null}
        selected={false}
        sizing="normal"
        stacked={false}
        style={Object {}}
        title="Next"
        totalItems={0}
        useedit={false}
        useinput={false}
        visible={true}
        widget={null}
        width=""
      />
      <ListItem
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className="test-class"
        color="inherit"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        hiddenLeftButton={false}
        hiddenRightButton={false}
        href={
          Object {
            "selectHandler": [Function],
            "sizing": "normal",
          }
        }
        id=""
        initialPage={1}
        initialPageSize={25}
        layout={5}
        leftButton={
          <Icon
            backgroundColor="inherit"
            borderColor="inherit"
            borderWidth="none"
            className=""
            color="inherit"
            contentEditable={false}
            direction="right"
            disabled={false}
            focus={false}
            height=""
            iconName=""
            id=""
            imageFile=""
            location="none"
            noedit={false}
            nohover={false}
            noripple={false}
            selected={false}
            sizing="normal"
            style={Object {}}
            visible={true}
            width=""
          />
        }
        location="none"
        noedit={true}
        nohover={false}
        noripple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onChangePageSize={[Function]}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseOut={[Function]}
        onSelect={[Function]}
        onSort={null}
        onUpdate={[Function]}
        pageSizes={
          Array [
            25,
            50,
            100,
          ]
        }
        pagesToDisplay={3}
        rightButton={null}
        selected={false}
        sizing="normal"
        stacked={false}
        style={Object {}}
        title="Previous"
        totalItems={0}
        useedit={false}
        useinput={false}
        visible={true}
        widget={null}
        width=""
      />
      <ListDivider
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className=""
        color="lightgray"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        hiddenLeftButton={false}
        hiddenRightButton={false}
        id=""
        layout={5}
        leftButton={null}
        location="none"
        noedit={false}
        nohover={false}
        noripple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseOut={[Function]}
        onUpdate={[Function]}
        rightButton={null}
        selected={false}
        sizing="normal"
        stacked={false}
        style={Object {}}
        title=""
        useedit={false}
        visible={true}
        widget={null}
        width=""
      />
      <ListItem
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className="test-class"
        color="inherit"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        hiddenLeftButton={false}
        hiddenRightButton={false}
        href={
          Object {
            "selectHandler": [Function],
            "sizing": "normal",
          }
        }
        id=""
        initialPage={1}
        initialPageSize={25}
        layout={5}
        leftButton={
          <Icon
            backgroundColor="inherit"
            borderColor="inherit"
            borderWidth="none"
            className=""
            color="green"
            contentEditable={false}
            direction="right"
            disabled={false}
            focus={false}
            height=""
            iconName="check"
            id=""
            imageFile=""
            location="none"
            noedit={false}
            nohover={false}
            noripple={false}
            selected={false}
            sizing="normal"
            style={Object {}}
            visible={true}
            width=""
          />
        }
        location="none"
        noedit={true}
        nohover={false}
        noripple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onChangePageSize={[Function]}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseOut={[Function]}
        onSelect={[Function]}
        onSort={null}
        onUpdate={[Function]}
        pageSizes={
          Array [
            25,
            50,
            100,
          ]
        }
        pagesToDisplay={3}
        rightButton={null}
        selected={false}
        sizing="normal"
        stacked={false}
        style={Object {}}
        title="25"
        totalItems={0}
        useedit={false}
        useinput={false}
        visible={true}
        widget={null}
        width=""
      />
      <ListItem
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className="test-class"
        color="inherit"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        hiddenLeftButton={false}
        hiddenRightButton={false}
        href={
          Object {
            "selectHandler": [Function],
            "sizing": "normal",
          }
        }
        id=""
        initialPage={1}
        initialPageSize={25}
        layout={5}
        leftButton={
          <Icon
            backgroundColor="inherit"
            borderColor="inherit"
            borderWidth="none"
            className=""
            color="inherit"
            contentEditable={false}
            direction="right"
            disabled={false}
            focus={false}
            height=""
            iconName=""
            id=""
            imageFile=""
            location="none"
            noedit={false}
            nohover={false}
            noripple={false}
            selected={false}
            sizing="normal"
            style={Object {}}
            visible={true}
            width=""
          />
        }
        location="none"
        noedit={true}
        nohover={false}
        noripple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onChangePageSize={[Function]}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseOut={[Function]}
        onSelect={[Function]}
        onSort={null}
        onUpdate={[Function]}
        pageSizes={
          Array [
            25,
            50,
            100,
          ]
        }
        pagesToDisplay={3}
        rightButton={null}
        selected={false}
        sizing="normal"
        stacked={false}
        style={Object {}}
        title="50"
        totalItems={0}
        useedit={false}
        useinput={false}
        visible={true}
        widget={null}
        width=""
      />
      <ListItem
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className="test-class"
        color="inherit"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        hiddenLeftButton={false}
        hiddenRightButton={false}
        href={
          Object {
            "selectHandler": [Function],
            "sizing": "normal",
          }
        }
        id=""
        initialPage={1}
        initialPageSize={25}
        layout={5}
        leftButton={
          <Icon
            backgroundColor="inherit"
            borderColor="inherit"
            borderWidth="none"
            className=""
            color="inherit"
            contentEditable={false}
            direction="right"
            disabled={false}
            focus={false}
            height=""
            iconName=""
            id=""
            imageFile=""
            location="none"
            noedit={false}
            nohover={false}
            noripple={false}
            selected={false}
            sizing="normal"
            style={Object {}}
            visible={true}
            width=""
          />
        }
        location="none"
        noedit={true}
        nohover={false}
        noripple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onChangePageSize={[Function]}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseOut={[Function]}
        onSelect={[Function]}
        onSort={null}
        onUpdate={[Function]}
        pageSizes={
          Array [
            25,
            50,
            100,
          ]
        }
        pagesToDisplay={3}
        rightButton={null}
        selected={false}
        sizing="normal"
        stacked={false}
        style={Object {}}
        title="100"
        totalItems={0}
        useedit={false}
        useinput={false}
        visible={true}
        widget={null}
        width=""
      />
      <ListItem
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className="test-class"
        color="inherit"
        contentEditable={false}
        direction="right"
        disabled={false}
        focus={false}
        height=""
        hiddenLeftButton={false}
        hiddenRightButton={false}
        href={
          Object {
            "selectHandler": [Function],
            "sizing": "normal",
          }
        }
        id=""
        initialPage={1}
        initialPageSize={25}
        layout={5}
        leftButton={
          <Icon
            backgroundColor="inherit"
            borderColor="inherit"
            borderWidth="none"
            className=""
            color="inherit"
            contentEditable={false}
            direction="right"
            disabled={false}
            focus={false}
            height=""
            iconName=""
            id=""
            imageFile=""
            location="none"
            noedit={false}
            nohover={false}
            noripple={false}
            selected={false}
            sizing="normal"
            style={Object {}}
            visible={true}
            width=""
          />
        }
        location="none"
        noedit={true}
        nohover={false}
        noripple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onChangePageSize={[Function]}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseOut={[Function]}
        onSelect={[Function]}
        onSort={null}
        onUpdate={[Function]}
        pageSizes={
          Array [
            25,
            50,
            100,
          ]
        }
        pagesToDisplay={3}
        rightButton={null}
        selected={false}
        sizing="normal"
        stacked={false}
        style={Object {}}
        title="all"
        totalItems={0}
        useedit={false}
        useinput={false}
        visible={true}
        widget={null}
        width=""
      />
    </List>
  </ButtonDialog>
</div>
`;

exports[`Test retrieval of Pager props object 1`] = `
Object {
  "backgroundColor": "inherit",
  "borderColor": "inherit",
  "borderWidth": "none",
  "className": "",
  "color": "inherit",
  "contentEditable": false,
  "direction": "right",
  "disabled": false,
  "focus": false,
  "height": "",
  "id": "",
  "initialPage": 1,
  "initialPageSize": 25,
  "location": "none",
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "onChangePageSize": [Function],
  "onSelect": [Function],
  "onSort": null,
  "pageSizes": Array [
    25,
    50,
    100,
  ],
  "pagesToDisplay": 3,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "totalItems": 0,
  "useinput": false,
  "visible": true,
  "width": "",
}
`;
